client/state/sites/domains/assembler.js

```diff
@@ -1,32 +1,36 @@
 /**
- * External dependencies
+ * Internal dependencies
  */
+import { getDomainType } from 'lib/domains/utils';
 
 export const createSiteDomainObject = domain => {
 	return {
 		autoRenewalDate: String( domain.auto_renewal_date ),
 		autoRenewing: Boolean( domain.auto_renewing ),
 		blogId: Number( domain.blog_id ),
 		canSetAsPrimary: Boolean( domain.can_set_as_primary ),
+		currentUserCanManage: Boolean( domain.current_user_can_manage ),
 		domain: String( domain.domain ),
-		pointsToWpcom: Boolean( domain.points_to_wpcom ),
 		expired: Boolean( domain.expired ),
-		expiry: String( domain.expiry ),
+		expiry: ! domain.expiry ? false : String( domain.expiry ),
```

Comment:
Agreed. However, I made this change to follow @retrofox's fixtures: https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/domains/test/fixture.js#L56-L56. Will fix. :)